### PR TITLE
Add support for Token Macro in spec. [Fixes JENKINS-46796]

### PR DESCRIPTION
**Pull request by mdkf**

### CODE REVIEW

**Comment by eyalbe4** [comment]
@mdkf,
Thank you for contributing this!
See my inline comments and let me know what you think.
Also, can you please join our contributors community by signing [JFrog's CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N) so that we can merge your code?

**Comment by mdkf** [comment]
Hi @eyalbe4 ,
 I cant see the inline comments.
-Michael

**Comment by eyalbe4** [comment]
Sorry about that @mdkf.
Looks like I didn't submit the review.
Can you see the comments now?

**Comment by eyalbe4** [comment]
@mdkf,
Would you like us to continue on this pull request?

**Comment by mdkf** [comment]
@eyalbe4 Have you had a chance to look at this?

**Comment by mdkf** [comment]
@eyalbe4 Have you had a chance to look at this?

File: src/main/java/org/jfrog/hudson/util/SpecUtils.java [hunk 1]
```diff
@@ -2,33 +2,54 @@
...
-
+        EnvVars env =build.getEnvironment(listener);
         if (StringUtils.isNotBlank(specConfiguration.getFilePath())) {
-            String filePath = specConfiguration.getFilePath().trim();
-            filePath = Util.replaceMacro(filePath, env);
-            String spec = buildDownloadSpecPath(filePath, workspace, logger).readToString();
-            return Util.replaceMacro(spec.trim(), env);
+            String filePath = expand(build, listener, specConfiguration.getFilePath().trim());
+            FilePath workspace= build.getExecutor().getCurrentWorkspace();
+            PrintStream logger =listener.getLogger();
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/util/SpecUtils.java]: Let's keep the spacing around the = sign according to the convention:
a = 1 (OK)
a =1 (not OK).
There are additional places in the code which should also be corrected.

**Comment by eyalbe4** [comment]
@mdkf,
Added some more inline comments and questions.
Also, can you please join our contributors community by signing [JFrog's CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N) so that we can merge your code?

**Comment by mdkf** [comment]
Sorry. I have just changed jobs and cannot work on it now.

**Pull request closed without merging**